plugins/bundle/plugin_test.go

```diff
@@ -318,35 +331,207 @@ func TestPluginListenerErrorClearedOn304(t *testing.T) {
 	b.Manifest.Init()
 
 	// Test that initial bundle is ok.
-	go plugin.oneShot(ctx, download.Update{Bundle: &b})
+	go plugin.oneShot(ctx, bundleName, download.Update{Bundle: &b})
 	s1 := <-ch
 
 	if s1.ActiveRevision != "quickbrownfaux" || s1.Code != "" {
 		t.Fatal("Unexpected status update, got:", s1)
 	}
 
 	// Test that service error triggers failure notification.
-	go plugin.oneShot(ctx, download.Update{Error: fmt.Errorf("some error")})
+	go plugin.oneShot(ctx, bundleName, download.Update{Error: fmt.Errorf("some error")})
 	s2 := <-ch
 
 	if s2.ActiveRevision != "quickbrownfaux" || s2.Code == "" {
 		t.Fatal("Unexpected status update, got:", s2)
 	}
 
 	// Test that service recovery triggers healthy notification.
-	go plugin.oneShot(ctx, download.Update{})
+	go plugin.oneShot(ctx, bundleName, download.Update{})
 	s3 := <-ch
 
 	if s3.ActiveRevision != "quickbrownfaux" || s3.Code != "" {
 		t.Fatal("Unexpected status update, got:", s3)
 	}
 }
 
+func TestPluginBulkListener(t *testing.T) {
+	ctx := context.Background()
+	manager := getTestManager()
+	plugin := Plugin{manager: manager, status: map[string]*Status{}, etags: map[string]string{}}
+	bundleNames := []string{
+		"b1",
+		"b2",
+		"b3",
+	}
+	for _, name := range bundleNames {
+		plugin.status[name] = &Status{Name: name}
+	}
+	bulkChan := make(chan map[string]*Status)
+
+	plugin.RegisterBulkListener("bulk test", func(status map[string]*Status) {
+		bulkChan <- status
+	})
+
+	module := "package gork\np[x] { x = 1 }"
+
+	b := bundle.Bundle{
+		Manifest: bundle.Manifest{
+			Revision: "quickbrownfaux",
+		},
+		Data: map[string]interface{}{},
+		Modules: []bundle.ModuleFile{
+			{
+				Path:   "/foo.rego",
+				Parsed: ast.MustParseModule(module),
+				Raw:    []byte(module),
+			},
+		},
+	}
+
+	b.Manifest.Init()
+
+	// Test that initial bundle is ok. Defer to separate goroutine so we can
+	// check result with channel.
+	go plugin.oneShot(ctx, bundleNames[0], download.Update{Bundle: &b})
+	s1 := <-bulkChan
+
+	s := s1[bundleNames[0]]
```

Comment:
Nit: The assertion logic is repeated below with slight differences in the offsets. I could have imagined factoring the logic into a separate function.

Reply:
I spent some time on that initially, but it ended up being somewhat complicated for (IMO) not a huge gain. If we used this same sort of check in more than a handful of spots I would see it as more beneficial but decided to axe it and just have some code duplication in the test. I'm happy to revisit this later though.